lib/presentation/display_qr_code/widgets/expiration_date_timer.dart

```diff
@@ -0,0 +1,84 @@
+import 'dart:async';
+
+import 'package:flutter/cupertino.dart';
+import 'package:superformula_mobile_test/application/display_qr_code/display_qr_code_bloc.dart';
+import 'package:superformula_mobile_test/domain/display_qr_code/value_objects/qr_seed_expiration_date.dart';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:superformula_mobile_test/presentation/core/misc/colors.dart';
+
+class QRCodeExpirationDateTimer extends StatefulWidget {
+  final QrSeedExpirationDate expirationDate;
+  const QRCodeExpirationDateTimer({
+    required this.expirationDate,
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  State<QRCodeExpirationDateTimer> createState() =>
+      _QRCodeExpirationDateTimerState();
+}
+
+class _QRCodeExpirationDateTimerState extends State<QRCodeExpirationDateTimer> {
+  late int _start;
+  late Timer _timer;
+
+  void startTimer() {
+    const oneSec = Duration(seconds: 1);
+    _timer = Timer.periodic(
+      oneSec,
+      (timer) {
+        if (_start <= 0) {
+          setState(() {
+            context
+                .read<DisplayQrCodeBloc>()
+                .add(const DisplayQrCodeEvent.requestedNewQrCode());
+            timer.cancel();
+          });
+        } else {
+          setState(() {
+            _start--;
+          });
+        }
+      },
+    );
+  }
+
+  @override
+  void initState() {
+    _start =
+        widget.expirationDate.getOrCrash().difference(DateTime.now()).inSeconds;
+    startTimer();
```

Comment:
bug: _start could be initialized to a negative number.

Reply:
I've changed (see newer commits) the way we are dealing with the expiration date of QR codes and caching, preventing `_start` from being initialized with a negative value. But there are no countermeasures to avoid it in the `expiration_date_timer.dart` yet. Is that what you meant?